### PR TITLE
Don't crash on diagnostics with no location

### PR DESCRIPTION
**Pull request by NTillmann**
Release notes: None

The CLI used to just ignore diagnostics without location, later triggering an invariant violation complaining that there were no diagnostics error in the presence of a FatalError exception. Thus was a really bad end-user experience. This pull request handles diagnostics that are missing locations.

### CODE REVIEW

**Comment by hermanventer** [comment]
I'm not particularly happy with this change. The AI engine does lots of inlining and unrolling and this may result in the same error message for the same line getting reported more than once. We should output only the first error for a particular line.  See test-internal.js for an alternative approach to dealing with errors that have no source locations.

**Comment by NTillmann** [comment]
Makes sense. Now that you pointed me at it, it's annoying to see so much code duplication where some issues already got addressed here but not there.

**Comment by hermanventer** [comment]
Agreed, I'm annoyed too. File an issue. :-)

File: src/prepack-cli.js [hunk 1]
```diff
@@ -267,15 +267,29 @@ function run(
   }
 
   let errors: Map<BabelNodeSourceLocation, CompilerDiagnostic> = new Map();
+  let errorList: Array<CompilerDiagnostic>;
```
Comment (hermanventer) [inline src/prepack-cli.js]: Initialize this to the empty list.

Comment (NTillmann) [reply src/prepack-cli.js]: I am impressed that we have a test for this :-)

**Pull request closed without merging**